tests/unit-tests/test_nn_model.py

```diff
@@ -224,6 +228,295 @@ def test_hgt_with_no_indegree_dstnodes(input_dim, output_dim):
+    src2, dst2, r1_eid = subg.edges(form='all', etype='r1')
+
+    src_idx = th.unique(th.concat([src1, src2]))
+    dst_idx = th.unique(th.concat([dst1, dst2]))
+
+    node_feats = {
+        "n0": th.rand(src_idx.shape[0], input_dim).to(dev),
+        "n1": th.rand(dst_idx.shape[0], input_dim).to(dev)
+    }
+    edge_feats = {
+        ("n0", "r0", "n1"): th.rand(r0_eid.shape[0], input_dim).to(dev),
+        ("n0", "r1", "n1"): th.rand(r1_eid.shape[0], input_dim).to(dev)
+    }
+
+    # Test case 0: normal case, have both node and edge feature on all node and edge types
+    layer = RelGraphConvLayer(
+        input_dim, output_dim, etypes,
+        num_bases=2,
+        edge_feat_name={("n0", "r0", "n1"): ['feat'], ("n0", "r1", "n1"): ['feat']},
+        activation=th.nn.ReLU(), self_loop=True,
+        dropout=0.1)
+    layer = layer.to(dev)
+
+    emb0 = layer(block, node_feats, edge_feats)
+    # check output numbers, dimensions and device
+    assert 'n0' not in emb0
+    assert emb0['n1'].shape[0] == len(seeds['n1'])
+    assert emb0['n1'].shape[1] == output_dim
+    assert emb0['n1'].get_device() == (-1 if dev == 'cpu' else 0)
+
+    # Test case 1: normal case, one edge type has features
+    layer = RelGraphConvLayer(
+        input_dim, output_dim, etypes,
+        num_bases=2,
+        edge_feat_name={("n0", "r0", "n1"): ['feat']},
+        activation=th.nn.ReLU(), self_loop=True,
+        dropout=0.1)
+    layer = layer.to(dev)
+
+    edge_feats = {
+        ("n0", "r0", "n1"): th.rand(r0_eid.shape[0], input_dim).to(dev)
+    }
+
+    emb1 = layer(block, node_feats, edge_feats)
+    # check output numbers, and dimensions
+    assert 'n0' not in emb1
+    assert emb1['n1'].shape[0] == len(seeds['n1'])
+    assert emb1['n1'].shape[1] == output_dim
+
+    layer = RelGraphConvLayer(
+        input_dim, output_dim, etypes,
+        num_bases=2,
+        edge_feat_name={("n0", "r1", "n1"): ['feat']},
+        activation=th.nn.ReLU(), self_loop=True,
+        dropout=0.1)
+    layer = layer.to(dev)
+
+    edge_feats = {
+        ("n0", "r1", "n1"): th.rand(r1_eid.shape[0], input_dim).to(dev)
+    }
+
+    emb1 = layer(block, node_feats, edge_feats)
+    # check output numbers, and dimensions
+    assert 'n0' not in emb1
+    assert emb1['n1'].shape[0] == len(seeds['n1'])
+    assert emb1['n1'].shape[1] == output_dim
+
+    # Test case 2: normal case, no model weights nor edge features as inputs
+    layer = RelGraphConvLayer(
+        input_dim, output_dim, etypes,
+        num_bases=2,
+        activation=th.nn.ReLU(), self_loop=True,
+        dropout=0.1)
+    layer = layer.to(dev)
+
+    emb2 = layer(block, node_feats)
+    # check output numbers, and dimensions
+    assert 'n0' not in emb2
+    assert emb2['n1'].shape[0] == len(seeds['n1'])
+    assert emb2['n1'].shape[1] == output_dim
+
+    # Test case 3: normal case, all 5 message passong ops
+    layer = RelGraphConvLayer(
```

Comment:
We should separately test GraphConvwithEdgeFeat with 5 message passing ops.

Reply:
separated as sub cases.

Reply:
It is better to check the output value.

Reply:
Test case 6 (line #425 to 516) checks the output values.